src/hotspot/share/runtime/arguments.cpp

```diff
@@ -3140,6 +3140,17 @@ jint Arguments::finalize_vm_init_args(bool patch_mod_javabase) {
     DynamicDumpSharedSpaces = true;
   }
 
+  if (AutoCreateSharedArchive) {
+    if (SharedArchiveFile == NULL) {
+      log_info(cds)("-XX:+AutoCreateSharedArchive must work with a valid SharedArchiveFile");
+      return JNI_ERR;
+    }
+    if (ArchiveClassesAtExit != NULL) {
+      log_info(cds)("-XX:+AutoCreateSharedArchive does not work with ArchiveClassesAtExit");
```

Comment:
These should be `log_warning()`. Otherwise the VM will exit without no apparent reason:
```
$ java -XX:+AutoCreateSharedArchive -version
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```